### PR TITLE
fail fast on connect error or non-zero return code

### LINKED ISSUE #139: Early exit if one/some host returns with non-zero error code
I have a scenario where commands returned with non-zero value in some hosts; but other hosts are still running the command.
Is there are option to early exit in this case ?

### PR DESCRIPTION
**Pull request by jerrymannil**
* Added "-k" option to enable fail fast option

* Fail fast on non-zero error code

Fixes #139

### CODE REVIEW

File: src/pdsh/opt.c [hunk 1]
```diff
@@ -66,7 +66,8 @@
 
 #define OPT_USAGE_DSH "\
 Usage: pdsh [-options] command ...\n\
--S                return largest of remote command return values\n"
+-S                return largest of remote command return values\n\
+-k                fail fast on ssh failures or non-zero return code\n"
```
Comment (grondo) [inline src/pdsh/opt.c:70]:
```suggestion
-k                fail fast on connect failures or non-zero return code\n"
```

Pdsh doesn't only support `ssh` so let's keep the option description generic.

**Review by grondo — COMMENTED** [review]
Thanks!

Just one minor suggestion inline.

Also, the manpage `doc/pdsh.1` needs to be updated, and it would be good to add at least one test to the testsuite.

**Comment by jerrymannil** [comment]
Thanks @grondo 
Review comments incorporated

**Review by grondo — COMMENTED** [review]
Nice, thanks!

Can you split the "Review comments incorporated" commit into a "doc:" commit and a "test:" commit? That way each commit does just one thing. Also, commits should have a body (something short is fine since it is clear what the commits are doing) This will satisfy the `validate commits` check in CI I think.

The new tests are failing in the macos builder, I'll try to determine why.

Thanks again!

**Comment by grondo** [comment]
The failures on macos were strange. I logging statement in `sshcmd_signal()` was causing `pdsh` to exit with code 64 instead of 1, which was causing the new tests of `-k` to fail. I have no idea why the call was failing, nor why the exit code was then 64. However, that statement was kind of redundant, so I've removed in a pending PR #146.

**Comment by jerrymannil** [comment]
@grondo  Thanks for the suggestions.  I have split the the commits.

**Comment by grondo** [comment]
#146 has been merged

**Comment by jerrymannil** [comment]
Merged fail-fast branch from master to pickup #146

**Comment by grondo** [comment]
Instead of merging from master, can you rebase, e.g `git rebase -i origin/master`? Thanks.

I think github wants approval for CI runs for first-time contributors, which is why tests did not rerun. After the rebase I'll approve another CI run. Thanks for your patience!

**Comment by jerrymannil** [comment]
@grondo rebase done

**Comment by grondo** [comment]
One last thing, when you `rebase -i` can you squash 304f4fa into 32a5f10. We typically don't leave review comment or fixup commits in the permanent history. Thanks!

Looks like all checks are passing now :tada: nice work!

**Comment by jerrymannil** [comment]
fixup commit squashed. 
Thanks @grondo for all the support.

**Review by grondo — APPROVED** [review]
LGTM! I'll set merge-when-passing label to get this merged. Thanks!

**Pull request merged**